src/validation/tests/array_validation_test.rs

```diff
@@ -105,3 +105,176 @@ fn array_access_dimension_mismatch() {
     assert_eq!(diagnostics.len(), 4);
     assert_validation_snapshot!(diagnostics);
 }
+
+#[test]
+fn assignment_1d() {
+    let diagnostics = parse_and_validate(
+        "
+		FUNCTION main : DINT
+			VAR
+				arr 		: ARRAY[1..5] OF DINT := [1, 2, 3, 4, 5, 6];
+				arr_alt		: ARRAY[1..5] OF DINT := (1, 2, 3, 4, 5, 6);
+			END_VAR
+
+			arr := [1, 2, 3, 4, 5]; // Valid
+			arr := [1, 2, 3, 4, 5, 6];
+			arr := (1, 2, 3, 4, 5, 6);
+		END_FUNCTION
+		",
+    );
+
+    assert_validation_snapshot!(diagnostics);
+}
+
+#[test]
+fn assignment_2d() {
+    let diagnostics = parse_and_validate(
+        "
+		FUNCTION main : DINT
+			VAR
+				arr 			: ARRAY[1..2, 1..5] OF DINT := [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11];
+				arr_alt			: ARRAY[1..2, 1..5] OF DINT := (1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11);
+
+				arr_nested		: ARRAY[1..2] OF ARRAY[1..5] OF DINT := [ [1, 2, 3, 4, 5], [6, 7, 8, 9, 10], [11, 12, 13, 14, 15] ];
+				arr_nested_alt	: ARRAY[1..2] OF ARRAY[1..5] OF DINT := ( [1, 2, 3, 4, 5], [6, 7, 8, 9, 10], [11, 12, 13, 14, 15] );
+			END_VAR
+
+			arr := [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]; // Valid
+			arr := [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11];
+			arr := (1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11);
+			
+			arr_nested		:= [ [1, 2, 3, 4, 5], [6, 7, 8, 9, 10] ]; // Valid
+			arr_nested		:= [ [1, 2, 3, 4, 5], [6, 7, 8, 9, 10], [11, 12, 13, 14, 15] ];
+			arr_nested		:= ( [1, 2, 3, 4, 5], [6, 7, 8, 9, 10], [11, 12, 13, 14, 15] );
+		END_FUNCTION
+		",
+    );
+
+    assert_validation_snapshot!(diagnostics);
+}
+
+#[test]
+fn assignment_3d() {
+    let diagnostics = parse_and_validate(
+        "
+		FUNCTION main : DINT
+			VAR
+				arr 		: ARRAY[1..2, 1..2, 1..2] OF DINT := [1, 2, 3, 4, 5, 6, 7, 8, 9];
+				arr_alt 	: ARRAY[1..2, 1..2, 1..2] OF DINT := (1, 2, 3, 4, 5, 6, 7, 8, 9);
+
+				arr_nested		: ARRAY[1..2] OF ARRAY[1..2] OF ARRAY[1..2] OF DINT := [ [[1, 2], [3, 4]], [[5, 6], [7, 8]], [[9, 10], [11, 12]] ];
+				arr_nested_alt	: ARRAY[1..2] OF ARRAY[1..2] OF ARRAY[1..2] OF DINT := ( [[1, 2], [3, 4]], [[5, 6], [7, 8]], [[9, 10], [11, 12]] );
+			END_VAR
+
+			arr := [1, 2, 3, 4, 5, 6, 7, 8]; // Valid
+			arr := [1, 2, 3, 4, 5, 6, 7, 8, 9];
+			arr := (1, 2, 3, 4, 5, 6, 7, 8, 9);
+			
+			arr_nested := [ [[1, 2], [3, 4]], [[5, 6], [7, 8]] ]; // Valid
+			arr_nested := [ [[1, 2], [3, 4]], [[5, 6], [7, 8]], [[9, 10], [11, 12]] ];
+			arr_nested := ( [[1, 2], [3, 4]], [[5, 6], [7, 8]], [[9, 10], [11, 12]] );
+		END_FUNCTION
+		",
+    );
+
+    assert_validation_snapshot!(diagnostics);
+}
+
+#[test]
+fn assignment_structs() {
+    let diagnostics = parse_and_validate(
+        "
+		TYPE FOO : STRUCT
+			idx : DINT;
+			arr : ARRAY[1..2] OF BAR;
+		END_STRUCT END_TYPE
+
+		TYPE BAR : STRUCT
+			arr : ARRAY[1..5] OF DINT;
+		END_STRUCT END_TYPE
+
+		FUNCTION main : DINT
+			VAR
```

Comment:
in here, add an assignment where an array is missing an element, an assignment with a missing field, and an out of order assignment for FOO